erpnext/crm/doctype/contract/contract.py

```diff
@@ -32,42 +40,41 @@ def update_contract_status(self):
 		else:
 			self.status = "Unsigned"
 
-	def set_contract_display(self):
-		self.contract_display = frappe.render_template(self.contract_terms, {"doc": self})
-
 	def update_fulfilment_status(self):
-		self.fulfilment_status = ""
+		fulfilment_status = "N/A"
 
 		if self.requires_fulfilment:
 			fulfilled_terms = self.get_fulfilled_terms()
 
 			if not fulfilled_terms:
-				self.fulfilment_status = "Unfulfilled"
+				fulfilment_status = "Unfulfilled"
 			elif fulfilled_terms < len(self.fulfilment_terms):
-				self.fulfilment_status = "Partially Unfulfilled"
+				fulfilment_status = "Partially Unfulfilled"
 			elif fulfilled_terms == len(self.fulfilment_terms):
-				self.fulfilment_status = "Fulfilled"
+				fulfilment_status = "Fulfilled"
 
 			if self.fulfilment_deadline:
 				now_date = getdate(nowdate())
 				deadline_date = getdate(self.fulfilment_deadline)
 
 				if now_date > deadline_date:
-					self.fulfilment_status = "Lapsed"
+					fulfilment_status = "Lapsed"
+
+		self.fulfilment_status = fulfilment_status
+
+	def set_contract_display(self):
+		self.contract_display = frappe.render_template(self.contract_terms, {"doc": self})
 
 	def get_fulfilled_terms(self):
```

Comment:
Name suggests that it actually returns the terms instead of the number of terms.